src/sortStudents.ts

```diff
@@ -1,16 +1,58 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'ask'|'desc';
 
+function getAvgOfGrades(student: Student): number {
+  return student.grades.reduce((acc: number, curr: number) => {
+    return acc + curr;
+  }, 0) / student.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy: Student[] = [...students];
+
+  return studentsCopy.sort((firstStudent: Student, secondStudent: Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+      case SortType.Married:
+        return order === 'desc'
+          ? secondStudent[sortBy].toString()
+            .localeCompare(firstStudent[sortBy].toString())
+          : firstStudent[sortBy].toString()
+            .localeCompare(secondStudent[sortBy].toString());
+
+      case SortType.Age:
+        return order === 'desc'
+          ? +secondStudent[sortBy] - +firstStudent[sortBy]
```

Comment:
```suggestion
          ? Number(secondStudent[sortBy]) - Number(firstStudent[sortBy])
```
Number() is more obvious way